dts/bindings/sensor/st,lis2dw12-common.yaml

```diff
@@ -104,6 +104,39 @@ properties:
       - 3
       - 4
 
+  fifo-mode:
+    type: int
+    default: 0
+    description: |
+      Specify the sensor FIFO mode. Default is 0 (Hardware register reset val)
+
+        0 # Bypass Mode. In this mode the FIFO is not operational no data
+        is collected in FIFO memory, and it remains empty with the only actual
+        sample available in the output registers.
+
+        1 # FIFO Mode. Data from the X, Y and Z channels are stored
+        in the FIFO until it is full, when 32 unread samples are stored
+        in memory, data collecting is stopped.
+
+        3 # Continuous-to-FIFO mode. FIFO operates in Continuous mode
+        and FIFO mode starts upon an internal trigger event. When the
+        FIFO is full, data collecting is stopped.
+
+        4 # Bypass-to-Continuous mode. Data measurement storage inside
+        FIFO starts in Continuous mode upon an internal trigger event,
+        then the sample that follows the trigger is available in FIFO.
+
+        6 # Continuous mode. This mode provides a continuous FIFO update:
+        when 32 unread samples are stored in memory, as new data arrives
+        the oldest data is discarded and overwritten by the newer.
+
+    enum:
+      - 0
+      - 1
+      - 3
+      - 4
+      - 6
```

Comment:
1. Is this change related to the commit itself? Or maybe should it be in a separate commit?
2. In any case, I don't see how you read the multiple data in FIFO. Shouldn't you change data fetch and channel get? And in which way?

Reply:
1. It is not directly related to the commit. I can keep the implementation of fifo-mode selection as a separate commit if wished.
2. Multiple data read is done internally within the driver's FIFO buffer. Here we can merely select which type of "multiple data read" we want. Data fetch and channel get do not need to be changed as the output of the FIFO buffer is still read from the same register address 
![Uploading image.png…]()

Reply:
> 1. It is not directly related to the commit. I can keep the implementation of fifo-mode selection as a separate commit if wished.
> 

Well, it's not my wish. I think it's mandatory to have atomic commits, with a single feature for each commit.
;)

>     2. Multiple data read is done internally within the driver's FIFO buffer. Here we can merely select which type of "multiple data read" we want. Data fetch and channel get do not need to be changed as the output of the FIFO buffer is still read from the same register address, regardless of the FIFO mode selection.

But let's say you receive an interrupt because you reached FIFO threshold (is this also checked in the driver?), then how can you read ALL data from the FIFO if you don't check number of FIFO items? Do you have a sample (which I guess would be nice to commit) to test it? Maybe you can prepare all this part in a separate PR

Reply:
SPOILER: I'm not sure we can handle FIFO in sensor API v1.

Reply:
I think I can move the fifo mode select change into a separate PR later. Would be satisfied for now to introduce the power domain handling only :)

Reply:
fifo mode select changes now discarded for this PR

Reply:
@avisconti FIFO mode could be supported if the driver moves to the newer API

Reply:
@SSaeedTonies I think you need to drop the CONTINUOS_MODE idea and propose a PR with only the POWER_MODE support